### PR TITLE
[Merged by Bors] - doc: Mark named theorems

### PR DESCRIPTION
**Pull request by hmonroe**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/FreeGroup/Basic.lean [hunk 1]
```diff
@@ -1343,7 +1343,7 @@ theorem toWord_inv {x : FreeGroup α} : x⁻¹.toWord = invRev x.toWord := by
 #align free_group.to_word_inv FreeGroup.toWord_inv
 #align free_add_group.to_word_neg FreeAddGroup.toWord_neg
 
-/-- Constructive Church-Rosser theorem (compare `church_rosser`). -/
+/-- **Constructive Church-Rosser theorem** (compare `church_rosser`). -/
 @[to_additive "Constructive Church-Rosser theorem (compare `church_rosser`)."]
```
Comment (eric-wieser) [inline Mathlib/GroupTheory/FreeGroup/Basic.lean:1347]:
```suggestion
@[to_additive "**Constructive Church-Rosser theorem** (compare `church_rosser`)."]
```

File: Mathlib/MeasureTheory/Function/Jacobian.lean [hunk 2]
```diff
@@ -598,8 +598,8 @@ theorem addHaar_image_eq_zero_of_differentiableOn_of_addHaar_eq_zero (hf : Diffe
     _ = 0 := by simp only [tsum_zero, mul_zero]
 #align measure_theory.add_haar_image_eq_zero_of_differentiable_on_of_add_haar_eq_zero MeasureTheory.addHaar_image_eq_zero_of_differentiableOn_of_addHaar_eq_zero
 
-/-- A version of Sard lemma in fixed dimension: given a differentiable function from `E` to `E` and
-a set where the differential is not invertible, then the image of this set has zero measure.
+/-- A version of **Sard lemma** in fixed dimension: given a differentiable function from `E` to `E`
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Function/Jacobian.lean:601]:
```suggestion
/-- A version of **Sard's lemma** in fixed dimension: given a differentiable function from `E` to `E`
```
maybe?

File: Mathlib/NumberTheory/Modular.lean [hunk 3]
```diff
@@ -435,7 +435,8 @@ theorem eq_zero_of_mem_fdo_of_T_zpow_mem_fdo {n : ℤ} (hz : z ∈ 𝒟ᵒ) (hg
     _ = 1 := add_halves 1
 #align modular_group.eq_zero_of_mem_fdo_of_T_zpow_mem_fdo ModularGroup.eq_zero_of_mem_fdo_of_T_zpow_mem_fdo
 
-/-- Any `z : ℍ` can be moved to `𝒟` by an element of `SL(2,ℤ)`  -/
+/-- **First Fundamental Domain Lemma**: Any `z : ℍ` can be moved to `𝒟` by an element of
```
Comment (alexjbest) [inline Mathlib/NumberTheory/Modular.lean:438]: I'm not sure this is really a named theorem is it?

Comment (hmonroe) [reply Mathlib/NumberTheory/Modular.lean]: OK I reverted that

File: Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean [hunk 4]
```diff
@@ -60,6 +60,7 @@ theorem Ico_map_valMinAbs_natAbs_eq_Ico_map_id (p : ℕ) [hp : Fact p.Prime] (a
     (inj_on_of_surj_on_of_card_le _ hmem hsurj le_rfl) hsurj
 #align zmod.Ico_map_val_min_abs_nat_abs_eq_Ico_map_id ZMod.Ico_map_valMinAbs_natAbs_eq_Ico_map_id
 
+/-- **Gauss' Lemma** -/
 private theorem gauss_lemma_aux₁ (p : ℕ) [Fact p.Prime] {a : ℤ}
```
Comment (alexjbest) [inline Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean:64]: surely this name should go on the non-private `gauss_lemma` below?

File: Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean [hunk 5]
```diff
@@ -115,6 +116,7 @@ theorem gauss_lemma {p : ℕ} [h : Fact p.Prime] {a : ℤ} (hp : p ≠ 2) (ha0 :
   simp_all [ne_neg_self hp one_ne_zero, (ne_neg_self hp one_ne_zero).symm]
 #align zmod.gauss_lemma ZMod.gauss_lemma
 
+/-- **Eisenstein's lemma** -/
```
Comment (alexjbest) [inline Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean:119]: same for this one

**Comment by alexjbest** [comment]
maintainer merge

**Comment by alexjbest** [comment]
PS I don't think this counts as "easy" which by definition means < 30s review time ;) it takes me more than 30s to scroll through all the changes even!

**Comment by hmonroe** [comment]
> PS I don't think this counts as "easy" which by definition means < 30s review time ;) it takes me more than 30s to scroll through all the changes even!

Fair enough! Thanks for reviewing

**Comment by hmonroe** [comment]
Oops I closed when I meant resolve

File: Mathlib/NumberTheory/Modular.lean [hunk 6]
```diff
@@ -435,7 +435,8 @@ theorem eq_zero_of_mem_fdo_of_T_zpow_mem_fdo {n : ℤ} (hz : z ∈ 𝒟ᵒ) (hg
     _ = 1 := add_halves 1
 #align modular_group.eq_zero_of_mem_fdo_of_T_zpow_mem_fdo ModularGroup.eq_zero_of_mem_fdo_of_T_zpow_mem_fdo
 
-/-- Any `z : ℍ` can be moved to `𝒟` by an element of `SL(2,ℤ)`  -/
+/-- First Fundamental Domain Lemma: Any `z : ℍ` can be moved to `𝒟` by an element of
```
[the remark this answers is not in the document]
Comment (hmonroe) [reply Mathlib/NumberTheory/Modular.lean:438]: The original commit made it bold, and Alex thought it did not qualify as a named theorem, which is fine with me.

File: Mathlib/Probability/Variance.lean [hunk 7]
```diff
@@ -278,7 +278,7 @@ theorem meas_ge_le_evariance_div_sq {X : Ω → ℝ} (hX : AEStronglyMeasurable
       ENNReal.rpow_one, evariance]
 #align probability_theory.meas_ge_le_evariance_div_sq ProbabilityTheory.meas_ge_le_evariance_div_sq
 
-/-- *Chebyshev's inequality* : one can control the deviation probability of a real random variable
+/-- Chebyshev's inequality: one can control the deviation probability of a real random variable
```
[the remark this answers is not in the document]
Comment (hmonroe) [reply Mathlib/Probability/Variance.lean]: OK I committed that--it is ok to mark Chebyshev's inequality as a named theorem in multiple places

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**